app/models/set_member_subject.rb

```diff
@@ -15,6 +16,7 @@ class SetMemberSubject < ActiveRecord::Base
   can_through_parent :subject_set, :update, :show, :destroy, :index, :update_links,
     :destroy_links
 
+  before_save :timestamp_newly_retired_workflows
```

Comment:
Yeah, you can pass a block to `update_links` in the controller [here](https://github.com/zooniverse/Panoptes/blob/master/lib/json_api_controller/updatable_resource.rb#L20) to perform custom work on the resource before the save, ala [this](https://github.com/zooniverse/Panoptes/blob/31316cab492cf2d5775f13b7e43bc6f790b38d54/app/controllers/api/v1/subject_sets_controller.rb#L13). Is this what you are looking for?